### PR TITLE
Overhaul `edit` feature in UG

### PR DESCRIPTION
**Pull request by stanley-1**
Consolidated `edit`-related features under [`edit INDEX [k/KEYWORDS]...`](https://stanley-1.github.io/tp/UserGuide.html#edit) and [`edit INDEX [k/]...`](https://stanley-1.github.io/tp/UserGuide.html#delete_fields) respectively.

Click on the links to access the UG and evaluate the changes on my repo.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -218,153 +215,148 @@ Scroll through your history of commands using the `↑` or `↓` arrow keys.
...
-* At least one of the optional fields must be provided.
-* Users can opt to change any fields associated with a contact as long as they prepend the argument with the corresponding flags according to the format above.
-* Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person’s tags by typing `t/` without
-    specifying any tag after it.
-* When editing dates, all existing dates of the person will be removed i.e adding of dates is not cumulative.
-* You can remove all the person’s dates by typing `date/` without
-  specifying any date after it.
+* You can provide the following fields (i.e. `[k/KEYWORDS]`) in any order when editing a contact's information: `[n/NAME] [p/PHONE] [r/REMARK] [t/TAG]…​ [date/NAME:YYYY-MM-DD[:monthly|:yearly]]…​ [fb/FACEBOOK] [ig/INSTAGRAM] [tele/TELEGRAM] [tiktok/TIKTOK] [twitter/TWITTER]`
```
Comment (david-eom) [inline docs/UserGuide.md:228]: Small nit: add period behind.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -218,153 +215,148 @@ Scroll through your history of commands using the `↑` or `↓` arrow keys.
...
+
+* Before you specify information for any field that you wish to modify, you should prepend your input with its corresponding flag (e.g. `n/` `p/` etc.) according to the format above.
+* You should check your input before you hit ENTER as SociaLite will overwrite the contact's existing information for any valid field that you furnish.
+* When you modify `TAG`s or `DATE`s, the existing `TAG`s or `DATE`s of the person will be removed i.e. adding of `TAG`s or `DATE`s is not cumulative.
+* Your input should conform with the character limit for the following fields:
+
+    | Field    | Character Limit |
+    | `TAG`    | 50              |
+    | `DATE`   | 50              |
+    | `REMARK` | 150             |
```
Comment (david-eom) [inline docs/UserGuide.md:245]: Nice!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -218,153 +215,148 @@ Scroll through your history of commands using the `↑` or `↓` arrow keys.
...
 
-Edits the phone number and Facebook handle of the 1st person to be `91234567` and `Yalex19` respectively.
-![05_edit](images/UG/05_edit.png)
 </div>
 
 <div markdown="block" class="alert alert-success">
-**:heavy_check_mark: Example:** `edit 4 n/Betsy Crower t/`
+**:heavy_check_mark: Example:** Edit `NAME`
```
Comment (david-eom) [inline docs/UserGuide.md:250]: I think the grave accent on this line, line 258 and line 266 can be a bit confusing to users! Maybe we can just get rid of the inline code and use "Edit phone number" "Edit contact name" "Edit a tag" instead.

**Review by david-eom — APPROVED** [review]
Hi Stanley, the changes to user guide looks good! Just a few small nits, I think changing it to the current format makes it much simpler for users to pick up. But any reason why `k` was picked haha?

**Comment by stanley-1** [comment]
> Hi Stanley, the changes to user guide looks good! Just a few small nits, I think changing it to the current format makes it much simpler for users to pick up. But any reason why `k` was picked haha?

Hey @david-eom, thanks for reviewing the PR. I've amended the areas you've pointed out and the changes are available on the same links above, so that you can have an easier review. I chose `k` since most of the flags usually start with the first letter of the parameter specified (e.g.: `n/NAME` or `fb/FACEBOOK`). But I can change it to any other format that you might have in mind! I decided to use `k/KEYWORDS` to be in line with the current approach and I couldn't really think of another way to encapsulate the diverse keywords/parameters just as how `p/PLATFORM` encapsulates all the social media handles. I guess we can consider `f/FIELDS` as an alternative? HAHA

**Comment by david-eom** [comment]
@stanley-1 Ohh i see haha, thanks for the explanation! then in that case `k/KEYWORD` works just as fine! My latest PR is also based on the `k/` format as well, so I think we can just keep it this way!

**Pull request merged**